.github/actions/clp-core-build-containers/action.yaml

```diff
@@ -35,58 +27,46 @@ runs:
         username: "${{github.actor}}"
         password: "${{inputs.token}}"
 
-    - id: "get_deps_image_props"
+    - id: "get_image_props"
       shell: "bash"
       run: |
+        name="clp-core-dependencies-x86-${{inputs.os_name}}"
         if [[ "${{inputs.push_deps_image}}" == "true" ]] ; then
-          echo "REGISTRY=ghcr.io" >> "$GITHUB_OUTPUT"
-          echo "BRANCH=${{github.ref_name}}" >> "$GITHUB_OUTPUT"
+          # Docker doesn't support repository names with uppercase characters, so we convert to
+          # lowercase here.
+          repository=$(echo '${{github.repository}}' | tr '[:upper:]' '[:lower:]')
+
+          echo "full_name=ghcr.io/${repository}/${name}" >> "$GITHUB_OUTPUT"
+          echo "branch=${{github.ref_name}}" >> "$GITHUB_OUTPUT"
+          echo "output=type=registry" >> "$GITHUB_OUTPUT"
         else
-          echo "REGISTRY=localhost:${{inputs.local_registry_port}}" >> "$GITHUB_OUTPUT"
-          echo "BRANCH=latest" >> "$GITHUB_OUTPUT"
+          image_path="/tmp/image.tar"
+          echo "image_path=${image_path}" >> "$GITHUB_OUTPUT"
+
+          echo "full_name=${name}" >> "$GITHUB_OUTPUT"
+          echo "branch=latest" >> "$GITHUB_OUTPUT"
+          echo "output=type=docker,dest=${image_path}" >> "$GITHUB_OUTPUT"
         fi
-        # Docker doesn't support repository names with uppercase characters, so we convert to
-        # lowercase here.
-        echo "REPOSITORY=$(echo '${{github.repository}}' | tr '[:upper:]' '[:lower:]')" \
-          >> "$GITHUB_OUTPUT"
 
     - id: "deps_image_meta"
       uses: "docker/metadata-action@v5"
       with:
-        images: "${{steps.get_deps_image_props.outputs.REGISTRY}}\
-          /${{steps.get_deps_image_props.outputs.REPOSITORY}}\
-          /clp-core-dependencies-x86-${{inputs.os_name}}"
+        images: "${{steps.get_image_props.outputs.full_name}}"
         tags: |
-          type=raw,value=${{steps.get_deps_image_props.outputs.BRANCH}}
+          type=raw,value=${{steps.get_image_props.outputs.branch}}
 
     - uses: "docker/build-push-action@v5"
       with:
         context: "${{inputs.clp_core_dir}}"
         file: "${{inputs.clp_core_dir}}/tools/docker-images/clp-env-base-${{inputs.os_name}}\
           /Dockerfile"
-        push: true
```

Comment:
Why isn't push not true anymore?

Reply:
From the [docs](https://github.com/docker/build-push-action#inputs):

> `push` is a shorthand for --output=type=registry (default false)

In the `push_deps_image == true` case, we're setting `output` to `type=registry` (equivalent to `push=true`). In the `push_deps_image == false` case, we want to output the image as a tar, so we set `output=type=docker,dest=<path>`.